model/database_cluster.go

```diff
@@ -0,0 +1,82 @@
+package model
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// DatabaseClusterInstallations is a container that holds a collection of installation IDs.
+type DatabaseClusterInstallations []string
```

Comment:
1/5 on this whole thought; take it or leave it :)

I'm not sure that this type is needed. The methods defined on it are basic CRUD (all but `Remove` have implementations that are very simple to understand) and don't do anything DatabaseClusterInstallation specific, and in addition, `DatabaseClusterInstallation` isn't a type so `DatabaseClusterInstallations` still has no guarantees -- nor does it make any itself -- about the type of string held by `DatabaseClusterInstallations`. 

So for instance, if I had some `databaseClusterInstallations []string` and I wanted to operate on that, I would find `len(databaseClusterInstallations)` in my calling routine equally clear or clearer than if I had some `databaseClusterInstallations *DatabaseClusterInstallation` and I performed `databaseClusterInstallations.Size()` except that I have to look up `Size` if I'm not familiar with this type, whereas `len()` is standard so I know that one off the top of my head.

If we're preparing to add methods to this type that take advantage of unique properties of the IDs held within this list, keeping this type makes sense to me, but otherwise it seems like unnecessary indirection with little benefit.

Reply:
I feel a bit more strongly that if you keep this type, though, it should be renamed to `DatabaseClusterInstallationIDs` because it does not hold pointers to `DatabaseClusterInstallation` objects, it holds a list of IDs, and so the name is misleading as-is.